openreview/conference/builder.py

```diff
@@ -0,0 +1,349 @@
+from __future__ import absolute_import
+
+import time
+from .. import openreview
+from .. import tools
+from . import webfield
+from . import invitation
+
+class ConferenceType(object):
+
+    @classmethod
+    def homepage_webfield_template(cls):
+        return "noBlindConferenceWebfield.js"
+
+    @classmethod
+    def submission_reply(cls, id, subject_areas, additional_fields):
+        return {}
+
+class SingleBlindConferenceType(ConferenceType):
+
+    @classmethod
+    def homepage_webfield_template(cls):
+        return "noBlindConferenceWebfield.js"
+
+class DoubleBlindConferenceType(ConferenceType):
+
+    @classmethod
+    def homepage_webfield_template(cls):
+        return "homepage.js"
+
+    @classmethod
+    def submission_reply(cls, id, subject_areas, additional_fields):
+
+        content = {
+            'title': {
+                'description': 'Title of paper.',
+                'order': 1,
+                'value-regex': '.{1,250}',
+                'required':True
+            },
+            'authors': {
+                'description': 'Comma separated list of author names. Please provide real names; identities will be anonymized.',
+                'order': 2,
+                'values-regex': "[^;,\\n]+(,[^,\\n]+)*",
+                'required':True
+            },
+            'authorids': {
+                'description': 'Comma separated list of author email addresses, lowercased, in the same order as above. For authors with existing OpenReview accounts, please make sure that the provided email address(es) match those listed in the author\'s profile. Please provide real emails; identities will be anonymized.',
+                'order': 3,
+                'values-regex': "([a-z0-9_\-\.]{2,}@[a-z0-9_\-\.]{2,}\.[a-z]{2,},){0,}([a-z0-9_\-\.]{2,}@[a-z0-9_\-\.]{2,}\.[a-z]{2,})",
+                'required':True
+            },
+            'keywords': {
+                'description': 'Comma separated list of keywords.',
+                'order': 6,
+                'values-regex': "(^$)|[^;,\\n]+(,[^,\\n]+)*"
+            },
+            'TL;DR': {
+                'description': '\"Too Long; Didn\'t Read\": a short sentence describing your paper',
+                'order': 7,
+                'value-regex': '[^\\n]{0,250}',
+                'required':False
+            },
+            'abstract': {
+                'description': 'Abstract of paper.',
+                'order': 8,
+                'value-regex': '[\\S\\s]{1,5000}',
+                'required':True
+            },
+            'pdf': {
+                'description': 'Upload a PDF file that ends with .pdf',
+                'order': 9,
+                'value-regex': 'upload',
+                'required':True
+            }
+        }
+
+        if subject_areas:
+            content['subject areas'] = {
```

Comment:
I don't remember that we defined that. Technically is possible. But this is a good time to make a decision.

Reply:
Yeah. My personal opinion is that field names should have one of the following patterns:

1) "Human Readable": fields are optimized for readability by formatting the string like a title, and can contain spaces.
2) "machine_readable": fields are optimized for compatibility by formatting the string in all lowercase and using underscores instead of spaces

right now it feels like we're somewhere in between